### PR TITLE
Update fuseki configuration notes

### PR DESCRIPTION
**Pull request by mgillis**
Add reasoner and tdb2 info

### CODE REVIEW

File: source/documentation/fuseki2/fuseki-configuration.md [hunk 1]
```diff
@@ -124,9 +124,71 @@ An in-memory dataset, with data in the default graph taken from a local file.
         ## tdb:unionDefaultGraph true ;
          .
 
+#### TDB2
+
+    <#dataset> rdf:type      tdb:DatasetTDB2 ;
+        tdb:location "/some/path" ;
+        # this can also have unionDefaultGraph, etc.
```
Comment (kinow) [inline source/documentation/fuseki2/fuseki-configuration.md:131]: Maybe use the same example above from TDB? Copying that part with the timeout (quite sure it's the same syntax, there's another example in [the docs](https://jena.apache.org/documentation/fuseki2/fuseki-config-endpoint.html)).

Same for `unionDefaultGraph`. Maybe just leave it commented out.

File: source/documentation/fuseki2/fuseki-configuration.md [hunk 2]
```diff
@@ -124,9 +124,71 @@ An in-memory dataset, with data in the default graph taken from a local file.
...
+An inference reasoner can be layered on top of a dataset as defined above. The type of reasoner must be selected carefully and should not include more reasoning than is required by the application, as extensive reasoning can be detrimental to performance.
+
+You have to build up layers of dataset, inference model, and graph.
+
+    <#dataset> rdf:type ja:RDFDataset;
+         ja:defaultGraph <#inferenceModel>
+         .
+         
+    <#inferenceModel> rdf:type      ja:InfModel;
+         ja:reasoner [ ja:reasonerURL <http://example/someReasonerURLHere> ];
```
Comment (kinow) [inline source/documentation/fuseki2/fuseki-configuration.md:145]: Maybe make a reference somewhere in the docs that this one here, is one of the values from the reasoners listed in the next section?

File: source/documentation/fuseki2/fuseki-configuration.md [hunk 3]
```diff
@@ -124,9 +124,71 @@ An in-memory dataset, with data in the default graph taken from a local file.
...
+         tdb2:location "/some/path/to/store/data/to";
+         # etc
+         .
+
+
+##### Possible reasoners:
+
+* **Generic Rule Reasoner**: `http://jena.hpl.hp.com/2003/GenericRuleReasoner`
+  
+  The specific rule set and mode configuration can be set either be method calls to the created reasoner or though parameters in the configuration Model.
```
Comment (kinow) [inline source/documentation/fuseki2/fuseki-configuration.md:158]: "can be set either be method"... by method?

"or though"... or through? (not a native speaker, so take it with a grain of salt)

**Comment by mgillis** [comment]
@kinow Thanks for taking a look. I agree this has some overlap with the other section, but I wasn't sure where to put this and it took me some digging to find it, I wanted to record it somewhere.

You're right about the typos and such -- I took the descriptions directly from the existing documentation in the code and didn't edit them. I can fix them up, just might not have time for a couple of days.

**Comment by afs** [comment]
Merged - thank you @mgillis.

I've also made changes inline with @kinow 's comments.

**Comment by mgillis** [comment]
@afs Oh, thanks! I apologize for not being able to get around to this in a reasonable time to incorporate the requested changes. I just had some time today and pulled this up and you've already done it. Thanks for your work on Jena.

**Comment by afs** [comment]
No problem - I'm clearing up in preparation for the 3.15.0 release .

**Pull request merged**